MPI_CXX_BOOL not defined in Microsoft MPI

When building with the MSVC compiler and Microsoft MPI, the following error occurs:
```
boost/mpi/datatype.hpp(217): error C2065: 'MPI_CXX_BOOL': undeclared identifier
```

`MPI_CXX_BOOL` is not defined in Microsoft MPI interface: https://learn.microsoft.com/en-us/message-passing-interface/mpi-datatype-enumeration

The problem has been introduced in commit 33765a1d301b1e096ad56d9d7f8080ffc178daf3.

Would it be possible to address this by using `MPI_C_BOOL` as an alternative for a boolean datatype? This might resolve the compatibility issue with Microsoft MPI.

@aminiussi Could you please look into this and provide guidance?

--- a/include/boost/mpi/datatype.hpp
+++ b/include/boost/mpi/datatype.hpp
@@ -213,8 +213,13 @@ BOOST_MPI_DATATYPE(packed, MPI_PACKED, builtin);
 BOOST_MPI_DATATYPE(char, MPI_CHAR, builtin);
 
 /// INTERNAL ONLY
+#if !defined(MSMPI_VER)
 /// We need to pick a boolean type, MPI_CXX_BOOL seems appropriate
 BOOST_MPI_DATATYPE(bool, MPI_CXX_BOOL, logical);
+#else
+/// but is not available on MSMPI
+BOOST_MPI_DATATYPE(bool, MPI_C_BOOL, logical);
+#endif
 
 /// INTERNAL ONLY
 BOOST_MPI_DATATYPE(short, MPI_SHORT, integer);

